ci/docker/centos-7-cpp.dockerfile

```
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM centos:centos7
+
+RUN yum install -y \
+        diffutils \
+        gcc-c++ \
+        libcurl-devel \
+        make \
+        openssl-devel \
+        wget \
+        which
+
+# yum install cmake version is too old
+
+ARG cmake=3.18.1

```

3.23.1 is newest, no reason to keep it pinned to something old.

I'm actually surprised this isn't defined in .env because this incantation exists in several dockerfiles, should we set the version there?

---

I was not sure if there was a reason to pin this version, that's why I did not update it, can do though.
The ubuntu 18.04 actually uses 3.10.2 as that is what apt has.

Yes I noticed that too, there are several different cmake versions pinned all over the project but I am not familiar enough with possible repercussions/advantages of bumping the version up to make a call on that.

---

You could upgrade and let CI tell you if it's fine 🤠 

Searching for `ARG cmake` I find three other instances: One with a comment pointing to an issue that has been fixed, one using the same version as that but with no comment, and another older one with no comment. We could trace those back through but I'd be willing to guess if there's no note explaining why it needs to be pinned on an older version, it's probably because that was the newest version at the time. Specifically for this case, coming from arrow-r-nightly, I can confirm that that's why it is on this version.

---

```suggestion
ARG cmake=3.23.1

```
I have made a note to look into the other cmake versions & places but it is out of scope for this PR.